packages/@tailwindcss-upgrade/src/template/codemods/automatic-var-injection.test.ts

```diff
@@ -0,0 +1,43 @@
+import { __unstable__loadDesignSystem } from '@tailwindcss/node'
+import { expect, test } from 'vitest'
+import { printCandidate } from '../candidates'
+import { automaticVarInjection } from './automatic-var-injection'
+
+test.each([
+  // Arbitrary candidates
+  ['[color:--my-color]', '[color:var(--my-color)]'],
+
+  // Arbitrary values for functional candidates
+  ['bg-[--my-color]', 'bg-[var(--my-color)]'],
+  ['bg-[color:--my-color]', 'bg-[color:var(--my-color)]'],
+  ['border-[length:--my-length]', 'border-[length:var(--my-length)]'],
+  ['border-[line-width:--my-width]', 'border-[line-width:var(--my-width)]'],
+
+  // Does not add var() if there is a _ before the variable name
+  ['bg-[_--my-color]', null],
+  ['bg-[color:_--my-color]', null],
+  ['border-[length:_--my-length]', null],
+  ['border-[line-width:_--my-width]', null],
+
+  // Modifiers
+  ['[color:--my-color]/[--my-opacity]', '[color:var(--my-color)]/[var(--my-opacity)]'],
+  ['bg-red-500/[--my-opacity]', 'bg-red-500/[var(--my-opacity)]'],
+  ['bg-[--my-color]/[--my-opacity]', 'bg-[var(--my-color)]/[var(--my-opacity)]'],
+  ['bg-[color:--my-color]/[--my-opacity]', 'bg-[color:var(--my-color)]/[var(--my-opacity)]'],
+
+  ['[color:--my-color]/[_--my-opacity]', '[color:var(--my-color)]/[_--my-opacity]'],
+  ['bg-red-500/[_--my-opacity]', null],
+  ['bg-[--my-color]/[_--my-opacity]', 'bg-[var(--my-color)]/[_--my-opacity]'],
+  ['bg-[color:--my-color]/[_--my-opacity]', 'bg-[color:var(--my-color)]/[_--my-opacity]'],
+
+  // Variants
+  ['supports-[--test]:flex', 'supports-[var(--test)]:flex'],
+  ['supports-[_--test]:flex', null],
```

Comment:
Ah, I got this from your list here: https://github.com/tailwindlabs/tailwindcss/pull/13657/files#diff-5c09a7c68a4426650ba1209e3be9652052750e24131acb607e09e5efea477dacR1619-R1624


`supports-[var(--test)]` mapping to `@supports var(--test)` seems to be at least compatible with v3 though 🙈 

<img width="886" alt="Screenshot 2024-09-26 at 18 19 27" src="https://github.com/user-attachments/assets/5a9a65ed-be15-473a-8769-1d7bfbfbed8d">

Reply:
Hmm I see. While this is compatible with v3, it's still wrong (aka will never match). So the question is:

1. Do we want to fix it (could be a breaking change since it's already broken right now)
2. Do we get rid of the supports variant in this case (or the candidate entirely because it won't do anything anyway)
3. Do we do what v4 expects, where it maps to `--test: var(--tw)` (which essentially is the fix from #1)
4. Don't do anything